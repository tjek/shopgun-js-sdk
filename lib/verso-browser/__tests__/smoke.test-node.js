const {JSDOM} = require('jsdom');
const {default: Verso} = require('../verso');

test("it doesn't break node", () => expect(true).beTruthy);

test('it can generate html with jsdom', () => {
    const {document} = new JSDOM(`
        <div class="verso">
            <div class="verso__scroller">
                <div class="verso__page-spread" data-id="page1" data-width="80">
                    <div class="verso__page">page1</div>
                </div>
                <div
                    class="verso__page-spread"
                    data-id="page2"
                    data-width="100"
                >
                    <div class="verso__page">page2</div>
                </div>
                <div class="verso__page-spread" data-id="page3" data-width="80">
                    <div class="verso__page">page3</div>
                </div>
            </div>
        </div>
    `).window;

    const verso = new Verso(document.querySelector('.verso')).start();
    verso.navigateTo(0, {duration: 0});
    expect(document.querySelector('.verso').outerHTML).toMatchSnapshot();
    verso.navigateTo(2, {duration: 0});
    expect(document.querySelector('.verso').outerHTML).toMatchSnapshot();
    verso.navigateTo(1, {duration: 0});
    expect(document.querySelector('.verso').outerHTML).toMatchSnapshot();
});
