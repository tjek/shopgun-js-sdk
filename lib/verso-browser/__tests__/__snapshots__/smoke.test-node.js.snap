// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it can generate html with jsdom 1`] = `
"<div class=\\"verso\\">
            <div class=\\"verso__scroller\\" style=\\"transition: none; transform: translateX(0%) translateY(0) scale(1);\\">
                <div class=\\"verso__page-spread\\" data-id=\\"page1\\" data-width=\\"80\\" data-active=\\"true\\" style=\\"left: 0%; display: block;\\">
                    <div class=\\"verso__page\\">page1</div>
                </div>
                <div class=\\"verso__page-spread\\" data-id=\\"page2\\" data-width=\\"100\\" style=\\"left: 80%; display: block;\\">
                    <div class=\\"verso__page\\">page2</div>
                </div>
                <div class=\\"verso__page-spread\\" data-id=\\"page3\\" data-width=\\"80\\">
                    <div class=\\"verso__page\\">page3</div>
                </div>
            </div>
        </div>"
`;

exports[`it can generate html with jsdom 2`] = `
"<div class=\\"verso\\">
            <div class=\\"verso__scroller\\" style=\\"transition: none; transform: translateX(-160%) translateY(0) scale(1);\\">
                <div class=\\"verso__page-spread\\" data-id=\\"page1\\" data-width=\\"80\\" data-active=\\"false\\" style=\\"left: 0%; display: none;\\">
                    <div class=\\"verso__page\\">page1</div>
                </div>
                <div class=\\"verso__page-spread\\" data-id=\\"page2\\" data-width=\\"100\\" style=\\"left: 80%; display: block;\\">
                    <div class=\\"verso__page\\">page2</div>
                </div>
                <div class=\\"verso__page-spread\\" data-id=\\"page3\\" data-width=\\"80\\" data-active=\\"true\\" style=\\"left: 180%; display: block;\\">
                    <div class=\\"verso__page\\">page3</div>
                </div>
            </div>
        </div>"
`;

exports[`it can generate html with jsdom 3`] = `
"<div class=\\"verso\\">
            <div class=\\"verso__scroller\\" style=\\"transition: none; transform: translateX(-80%) translateY(0) scale(1);\\">
                <div class=\\"verso__page-spread\\" data-id=\\"page1\\" data-width=\\"80\\" data-active=\\"false\\" style=\\"left: 0%; display: block;\\">
                    <div class=\\"verso__page\\">page1</div>
                </div>
                <div class=\\"verso__page-spread\\" data-id=\\"page2\\" data-width=\\"100\\" style=\\"left: 80%; display: block;\\" data-active=\\"true\\">
                    <div class=\\"verso__page\\">page2</div>
                </div>
                <div class=\\"verso__page-spread\\" data-id=\\"page3\\" data-width=\\"80\\" data-active=\\"false\\" style=\\"left: 180%; display: block;\\">
                    <div class=\\"verso__page\\">page3</div>
                </div>
            </div>
        </div>"
`;
